runhouse/resources/hardware/cluster.py

```diff
@@ -588,10 +594,56 @@ def on_this_cluster(self):
 
     # ----------------- RPC Methods ----------------- #
 
-    def call_client_method(self, method_name, *args, **kwargs):
-        self.check_server()
-        method = getattr(self.client, method_name)
-        return method(*args, **kwargs)
+    def call_client_method(self, method_name, *args, restart_server=True, **kwargs):
+        def check_and_call():
+            try:
+                # setup the client connection, and check server in background call
+                client = self.client
+                check_call = threading.Thread(target=client.check_server)
```

Comment:
how does the background call help stuff?

Reply:
it's just doing a background call to check_server at the same time as the method call rather than sequentially. if that throws an error itll be propagated back up